### ISSUE TITLE
Issue with the version of the pcmdi_metrics repo?

### ISSUE THREAD

**Issue by jservonnat** [PCMDI/pcmdi_metrics#288]
@doutriaux1 @durack1 Last week I've successfully installed the latest version of the PMP here at IPSL, using the latest install_metrics.bash script available. But the pcmdi_metrics_driver.py I get by running the install script does not make use of unidata (for the unity check/management).
But if I get directly the pcmdi_metrics repo from github (git clone git://github.com/PCMDI/pcmdi_metrics.git), the pcmdi_metrics_driver.py file I get does use unidata:
```
# Before we do anything else we need to create some units
# Salinity Units
import unidata
unidata.udunits_wrap.init()

# Create a dimensionless units named dimless
unidata.addDimensionlessUnit("dimless")

# Created scaled units for dimless
unidata.addScaledUnit("psu", .001, "dimless")
unidata.addScaledUnit("PSS-78", .001, "dimless")
unidata.addScaledUnit("Practical Salinity Scale 78", .001, "dimless")
```
Lastly, if I clone the sources in tmp/metrics of the successful install, and run "python setup.py install", the the setup looks successful but the test fails:
```
test_from_parameter_file (test_from_param.TestFromParam) ... Traceback (most recent call last):
  File "/data/jservon/PMP_installs/2015-11-06/PCMDI_METRICS/bin/pcmdi_metrics_driver.py", line 26, in <module>
    unidata.addDimensionlessUnit("dimless")
AttributeError: 'module' object has no attribute 'addDimensionlessUnit'
ok
```
(well, it says ok but there is an error and it displays no result)

So how can I get the version of the PMP using unidat installed properly?
Thanks!

**Comment by durack1** [comment]
@jservonnat it seems to me that you're not cloning the `master` branch correctly, as this version of [`pcmdi_metrics_driver.py` does in fact include the above](https://github.com/PCMDI/pcmdi_metrics/blob/master/src/python/pcmdi/scripts/pcmdi_metrics_driver.py#L20-L23)

We have a couple of tweaks which should be merged into the `master` branch tomorrow - just have to work with @doutriaux1 to get these in, and then you should be able to try again and successfully install the `master` version..

To make sure you're on `master` use this command:
```
[durack1@oceanonly pcmdi_metrics]$ git log --pretty=format:'%h' -n 1
3100dca
```

This should match the current master SHA1 `3100dca` as found on the [`pcmdi_metrics` repo homepage](https://github.com/PCMDI/pcmdi_metrics)
![3100dca](https://cloud.githubusercontent.com/assets/3229632/11083777/2dacd3be-87e8-11e5-9515-6751e6d1f091.png)

If you're not you'll need to issue a:
```
[durack1@oceanonly pcmdi_metrics]$ git fetch --all -p
Fetching origin
remote: Counting objects: 31, done.
remote: Compressing objects: 100% (31/31), done.
remote: Total 31 (delta 18), reused 0 (delta 0), pack-reused 0
Unpacking objects: 100% (31/31), done.
From github.com:PCMDI/pcmdi_metrics
   063ad34..69873c5  climo_script -> origin/climo_script
 * [new branch]      issue_282_tracking -> origin/issue_282_tracking
 * [new branch]      issue_284_matplotlib -> origin/issue_284_matplotlib
[durack1@oceanonly pcmdi_metrics]$ git pull
Already up-to-date.
```
And you're up to date with `master`

**Comment by durack1** [comment]
@jservonnat let us know if that solves it and I'll close this issue..

**Comment by jservonnat** [comment]
@durack1 I've tried an install today (wget the install_metrics.bash script, and run it without changing anything), and this is what I got:
```
jservon@ciclad-ng:/data/jservon/PMP_installs/2015-11-13/PCMDI_METRICS/tmp/metrics> git log --pretty=format:'%h' -n 1
git: /data/jservon/PMP_installs/2015-11-13/PCMDI_METRICS/Externals/lib/libz.so.1: no version information available (required by git)
11d71de
```
I've tried to fetch the origin as you explain:
```
jservon@ciclad-ng:/data/jservon/PMP_installs/2015-11-13/PCMDI_METRICS/tmp/metrics> git fetch --all -p
git: /data/jservon/PMP_installs/2015-11-13/PCMDI_METRICS/Externals/lib/libz.so.1: no version information available (required by git)
Fetching origin
git: /data/jservon/PMP_installs/2015-11-13/PCMDI_METRICS/Externals/lib/libz.so.1: no version information available (required by git)
```
And then:
```
jservon@ciclad-ng:/data/jservon/PMP_installs/2015-11-13/PCMDI_METRICS/tmp/metrics> git pull
git: /data/jservon/PMP_installs/2015-11-13/PCMDI_METRICS/Externals/lib/libz.so.1: no version information available (required by git)
git: /data/jservon/PMP_installs/2015-11-13/PCMDI_METRICS/Externals/lib/libz.so.1: no version information available (required by git)
git: /data/jservon/PMP_installs/2015-11-13/PCMDI_METRICS/Externals/lib/libz.so.1: no version information available (required by git)
git: /data/jservon/PMP_installs/2015-11-13/PCMDI_METRICS/Externals/lib/libz.so.1: no version information available (required by git)
git: /data/jservon/PMP_installs/2015-11-13/PCMDI_METRICS/Externals/lib/libz.so.1: no version information available (required by git)
git: /data/jservon/PMP_installs/2015-11-13/PCMDI_METRICS/Externals/lib/libz.so.1: no version information available (required by git)
git: /data/jservon/PMP_installs/2015-11-13/PCMDI_METRICS/Externals/lib/libz.so.1: no version information available (required by git)
git: /data/jservon/PMP_installs/2015-11-13/PCMDI_METRICS/Externals/lib/libz.so.1: no version information available (required by git)
git: /data/jservon/PMP_installs/2015-11-13/PCMDI_METRICS/Externals/lib/libz.so.1: no version information available (required by git)
git: /data/jservon/PMP_installs/2015-11-13/PCMDI_METRICS/Externals/lib/libz.so.1: no version information available (required by git)
git: /data/jservon/PMP_installs/2015-11-13/PCMDI_METRICS/Externals/lib/libz.so.1: no version information available (required by git)
You are not currently on a branch, so I cannot use any
'branch.<branchname>.merge' in your configuration file.
Please specify which remote branch you want to use on the command
line and try again (e.g. 'git pull <repository> <refspec>').
See git-pull(1) for details.
```
I suppose that there might be a way for me to make it work on my side but you might also be interested by those feedbacks, so I'm reporting my basic use of the install script as if I was totally new to this game...

**Comment by durack1** [comment]
@jservonnat yep this is what you'd expect when you haven't cloned the repo (and the hidden .git directory and all the interesting git stuff) and just downloaded the single file outside of git..

When you issue a:
`git clone https://github.com/PCMDI/pcmdi_metrics.git`

You should then get a (lower case) `pcmdi_metrics` directory with all the source, and if you do a `git log` within that directory (with it's `.git` and other bits) it'll then report your `SHA1` and what branch you're on..

**Comment by durack1** [comment]
@jservonnat and the [very top of your comment above](https://github.com/PCMDI/pcmdi_metrics/issues/288#issuecomment-156535836) you have a `SHA1` as `11d71de`.

Now that relates to the 13th January 2015 `v1.0.0` release.. Which is findable at https://github.com/PCMDI/pcmdi_metrics/tree/v1.0.0

**Comment by jservonnat** [comment]
@durack1 Ok, then I might need to learn to control the version from the install script. If you have the solution I take it ;)

**Comment by durack1** [comment]
@jservonnat for completeness the `unidata` stuff linked at the top, was included on the 17th April 2015 in https://github.com/PCMDI/pcmdi_metrics/commit/14648146a4dc9ec04e5027468c99bbfbaab4c7a4 with `SHA1` `1464814`

**Comment by durack1** [comment]
@jservonnat stackoverflow.com is your new best friend http://stackoverflow.com/questions/3555107/git-clone-particular-version-of-remote-repository

**Comment by jservonnat** [comment]
@durack1 Ok so I've commented the 'git checkout ${metrics_checkout}' line and now it's fine.
I'll check tomorrow if I can merge my IPSL developments. Thank you for your help!

**Comment by jservonnat** [comment]
Thanks @durack1 ;)

**Comment by durack1** [comment]
@jservonnat you can control this a little better by changing the values at https://github.com/PCMDI/pcmdi_metrics/blob/master/install_metrics.bash#L386-L390 - these set either the `v1.0.0` or `master` versions of the `metrics` along with also controlling the version of `UV-CDAT/CDAT` that is installed.. Our current latest `master` branch uses `master` for both

**Comment by jservonnat** [comment]
@durack1 I'm afraid it's not over...
I've set cdat_version="master" and metrics_checkout="master", and this is what I got:
```
[...]
Removing /data/jservon/PMP_installs/2015-11-16/PCMDI_METRICS/lib/python2.7/site-packages/cdutil-_-py2.7.egg-info
Writing /data/jservon/PMP_installs/2015-11-16/PCMDI_METRICS/lib/python2.7/site-packages/cdutil-_-py2.7.egg-info


*******************************
UVCDAT  - master - Install Success
Metrics - master - Install Success
*******************************
Please test as follows:
source /data/jservon/PMP_installs/2015-11-16/PCMDI_METRICS/bin/setup_runtime.sh or .csh
python /data/jservon/PMP_installs/2015-11-16/PCMDI_METRICS/tmp/metrics/test/test_suite.py
*******************************
Create your customized input_parameters.py (inspire yourself from examples in /data/jservon/PMP_installs/2015-11-16/PCMDI_METRICS/doc/pcmdi_input_parameters_sample.py
Once you have a parameter file run:
source /data/jservon/PMP_installs/2015-11-16/PCMDI_METRICS/bin/setup_runtime.sh
pcmdi_metrics_driver.py -p /path/to/your/edited/parameter_file.py
*******************************
Once everything is ok, you can safely remove the temporary directory: /data/jservon/PMP_installs/2015-11-16/PCMDI_METRICS/tmp
*******************************
For further information or suggestions please contact the PCMDI Metrics Team @ pcmdi-metrics@llnl.gov
*******************************

jservon@ciclad-ng:/data/jservon/PMP_installs/2015-11-16> source /data/jservon/PMP_installs/2015-11-16/PCMDI_METRICS/bin/setup_runtime.sh
Successfully updated your environment to use UVCDAT
(changes are valid for this session/terminal only)
Version: 2.4.0.rc2-100-gfa0d2e2
Location: /data/jservon/PMP_installs/2015-11-16/PCMDI_METRICS
jservon@ciclad-ng:/data/jservon/PMP_installs/2015-11-16> python /data/jservon/PMP_installs/2015-11-16/PCMDI_METRICS/tmp/metrics/test/test_suite.py
ERROR import vcs, skipping graphics test...
flake8 (test_flake8.TestFlake8) ... ok
test_from_parameter_file (test_from_param.TestFromParam) ...  parameter file ref is:  ['all']
 refs: [u'default']
 ref is:  [u'default']
/data/jservon/PMP_installs/2015-11-16/PCMDI_METRICS/lib/python2.7/site-packages/ESMP/src/ESMP_API.py:1241: FutureWarning: comparison to `None` will result in an elementwise object comparison in the future.
  if (srcMaskValues != None):
/data/jservon/PMP_installs/2015-11-16/PCMDI_METRICS/lib/python2.7/site-packages/ESMP/src/ESMP_API.py:1248: FutureWarning: comparison to `None` will result in an elementwise object comparison in the future.
  if (dstMaskValues != None):
 OBS SHAPE IS  (12, 72, 144)
 tos   GFDL-ESM2G   (12, 72, 144)   (12, 72, 144)    default
/data/jservon/PMP_installs/2015-11-16/PCMDI_METRICS/lib/python2.7/site-packages/numpy/ma/core.py:935: RuntimeWarning: overflow encountered in multiply
  result = self.f(da, db, *args, **kwargs)
 parameter file ref is:  ['all']
 refs: [u'default', u'alternate1', u'alternate2']
 ref is:  [u'default', u'alternate1', u'alternate2']
 OBS SHAPE IS  (12, 72, 144)
 tas   GFDL-ESM2G   (12, 72, 144)   (12, 72, 144)    default
Fatal Python error: Couldn't create autoTLSkey mapping
comparing: pcmdi_install_test_results/metrics_results/installationTest/tos_2.5x2.5_esmf_linear_metrics.json /data/jservon/PMP_installs/2015-11-16/PCMDI_METRICS/tmp/metrics/test/pcmdi/installationTest/tos_2.5x2.5_esmf_linear_metrics.json
Failing line: -                    "bias_xy_djf_GLB": "0.29",

Failing line: -                    "bias_xy_djf_SHEX": "2.26",

Failing line: -                    "bias_xy_jja_SHEX": "0.70",

Failing line: -                    "bias_xy_mam_GLB": "0.13",

Failing line: -                    "bias_xy_mam_NHEX": "-0.95",

Failing line: -                    "bias_xy_son_NHEX": "-1.18",

Failing line: -                    "bias_xy_son_SHEX": "0.67",

Failing line: -                    "mae_xy_djf_NHEX": "1.30",

Failing line: -                    "mae_xy_jja_NHEX": "1.11",

Failing line: -                    "mae_xy_mam_NHEX": "1.18",

Failing line: -                    "mae_xy_son_NHEX": "1.32",

Failing line: -                    "rms_xy_ann_NHEX": "1.87",

Failing line: -                    "rms_xy_ann_SHEX": "1.61",

Failing line: -                    "rms_xy_djf_GLB": "1.83",

Failing line: -                    "rms_xy_djf_NHEX": "2.18",

Failing line: -                    "rms_xy_jja_TROPICS": "1.19",

Failing line: -                    "rms_xy_mam_NHEX": "1.98",

Failing line: -                    "rms_xy_son_NHEX": "2.09",

Failing line: -                    "rms_xy_son_TROPICS": "1.28",

Failing line: -                    "rms_xyt_ann_NHEX": "2.05",

Failing line: -                    "rms_xyt_ann_TROPICS": "1.30",

FAIL
test_from_parameter_file (test_from_param.TestFromParam) ...  parameter file ref is:  ['default']
 ref is:  ['default']
/data/jservon/PMP_installs/2015-11-16/PCMDI_METRICS/lib/python2.7/site-packages/ESMP/src/ESMP_API.py:1241: FutureWarning: comparison to `None` will result in an elementwise object comparison in the future.
  if (srcMaskValues != None):
/data/jservon/PMP_installs/2015-11-16/PCMDI_METRICS/lib/python2.7/site-packages/ESMP/src/ESMP_API.py:1248: FutureWarning: comparison to `None` will result in an elementwise object comparison in the future.
  if (dstMaskValues != None):
 OBS SHAPE IS  (12, 72, 144)
 tas   GFDL-ESM2G   (12, 72, 144)   (12, 72, 144)    default
 Error while processing variable tas:
	not all arguments converted during string formatting
ok
test_from_parameter_file (test_from_param.TestFromParam) ... CDMS system error: No such file or directory
CDMS I/O error: Opening file /data/jservon/PMP_installs/2015-11-16/PCMDI_METRICS/tmp/metrics/test/pcmdi/data/sftflf_not_here_GFDL-ESM2G.nc
 No mask for  /data/jservon/PMP_installs/2015-11-16/PCMDI_METRICS/tmp/metrics/test/pcmdi/data/sftflf_not_here_GFDL-ESM2G.nc
 parameter file ref is:  ['default']
 ref is:  ['default']
/data/jservon/PMP_installs/2015-11-16/PCMDI_METRICS/lib/python2.7/site-packages/ESMP/src/ESMP_API.py:1241: FutureWarning: comparison to `None` will result in an elementwise object comparison in the future.
  if (srcMaskValues != None):
/data/jservon/PMP_installs/2015-11-16/PCMDI_METRICS/lib/python2.7/site-packages/ESMP/src/ESMP_API.py:1248: FutureWarning: comparison to `None` will result in an elementwise object comparison in the future.
  if (dstMaskValues != None):
 OBS SHAPE IS  (12, 72, 144)
 tas   GFDL-ESM2G   (12, 72, 144)   (12, 72, 144)    default
 Error while processing variable tas:
	not all arguments converted during string formatting
ok
test_from_parameter_file (test_from_param.TestFromParam) ... CDMS system error: No such file or directory
CDMS I/O error: Opening file /data/jservon/PMP_installs/2015-11-16/PCMDI_METRICS/tmp/metrics/test/pcmdi/data/sftflf_not_here_GFDL-ESM2G.nc
 No mask for  /data/jservon/PMP_installs/2015-11-16/PCMDI_METRICS/tmp/metrics/test/pcmdi/data/sftflf_not_here_GFDL-ESM2G.nc
 parameter file ref is:  ['default']
 ref is:  ['default']
/data/jservon/PMP_installs/2015-11-16/PCMDI_METRICS/lib/python2.7/site-packages/ESMP/src/ESMP_API.py:1241: FutureWarning: comparison to `None` will result in an elementwise object comparison in the future.
  if (srcMaskValues != None):
/data/jservon/PMP_installs/2015-11-16/PCMDI_METRICS/lib/python2.7/site-packages/ESMP/src/ESMP_API.py:1248: FutureWarning: comparison to `None` will result in an elementwise object comparison in the future.
  if (dstMaskValues != None):
 OBS SHAPE IS  (12, 72, 144)
 tas   GFDL-ESM2G   (12, 72, 144)   (12, 72, 144)    default
 Error while processing variable tas:
	not all arguments converted during string formatting
ok
test_from_parameter_file (test_from_param.TestFromParam) ... CDMS system error: No such file or directory
CDMS I/O error: Opening file /data/jservon/PMP_installs/2015-11-16/PCMDI_METRICS/tmp/metrics/test/pcmdi/data/sftlf_GFDL-ESM2G_fx_historical_r0i0p0_198501-200512-clim.nc
 No mask for  /data/jservon/PMP_installs/2015-11-16/PCMDI_METRICS/tmp/metrics/test/pcmdi/data/sftlf_GFDL-ESM2G_fx_historical_r0i0p0_198501-200512-clim.nc
 Error while processing variable aVarThatIsNot:
	'aVarThatIsNot'
 parameter file ref is:  ['default']
 ref is:  ['default']
/data/jservon/PMP_installs/2015-11-16/PCMDI_METRICS/lib/python2.7/site-packages/ESMP/src/ESMP_API.py:1241: FutureWarning: comparison to `None` will result in an elementwise object comparison in the future.
  if (srcMaskValues != None):
/data/jservon/PMP_installs/2015-11-16/PCMDI_METRICS/lib/python2.7/site-packages/ESMP/src/ESMP_API.py:1248: FutureWarning: comparison to `None` will result in an elementwise object comparison in the future.
  if (dstMaskValues != None):
 OBS SHAPE IS  (12, 72, 144)
 tos   GFDL-ESM2G   (12, 72, 144)   (12, 72, 144)    default
/data/jservon/PMP_installs/2015-11-16/PCMDI_METRICS/lib/python2.7/site-packages/numpy/ma/core.py:935: RuntimeWarning: overflow encountered in multiply
  result = self.f(da, db, *args, **kwargs)
CDMS system error: No such file or directory
CDMS I/O error: Opening file /data/jservon/PMP_installs/2015-11-16/PCMDI_METRICS/tmp/metrics/test/pcmdi/obs/fx/sftlf/UKMETOFFICE-HadISST-v1-1/sftlf_pcmdi-metrics_fx_UKMETOFFICE-HadISST-v1-1_198002-200501-clim.nc
 Could not find obs mask, generating
/data/jservon/PMP_installs/2015-11-16/PCMDI_METRICS/lib/python2.7/site-packages/cdms2/avariable.py:1088: Warning:
avariable.regrid: We chose regridMethod = linear for you among the following choices:
    'conserve' or 'linear' or 'patch'
  warnings.warn(message, Warning)
 OBS SHAPE IS  (12, 72, 144)
 Error while processing variable tos:
	not all arguments converted during string formatting
ok
test_from_parameter_file (test_from_param.TestFromParam) ...  parameter file ref is:  ['ERAINT']
 ref is:  ['ERAINT']
 OBS SHAPE IS  (12, 72, 144)
 tas   GFDL-ESM2G   (12, 72, 144)   (12, 72, 144)    ERAINT
 OBS SHAPE IS  (12, 72, 144)
 tas   GFDL-ESM2G   (12, 72, 144)   (12, 72, 144)    ERAINT
Fatal Python error: Couldn't create autoTLSkey mapping
ok
test_from_parameter_file (test_from_param.TestFromParam) ... CDMS system error: No such file or directory
CDMS I/O error: Opening file /data/jservon/PMP_installs/2015-11-16/PCMDI_METRICS/tmp/metrics/test/pcmdi/data/sftlf_GFDL-CM3_fx_historical_000101-010012-clim.nc
 No mask for  /data/jservon/PMP_installs/2015-11-16/PCMDI_METRICS/tmp/metrics/test/pcmdi/data/sftlf_GFDL-CM3_fx_historical_000101-010012-clim.nc
 parameter file ref is:  ['JPL-Aquarius-v2']
 ref is:  ['JPL-Aquarius-v2']
/data/jservon/PMP_installs/2015-11-16/PCMDI_METRICS/lib/python2.7/site-packages/ESMP/src/ESMP_API.py:1241: FutureWarning: comparison to `None` will result in an elementwise object comparison in the future.
  if (srcMaskValues != None):
/data/jservon/PMP_installs/2015-11-16/PCMDI_METRICS/lib/python2.7/site-packages/ESMP/src/ESMP_API.py:1248: FutureWarning: comparison to `None` will result in an elementwise object comparison in the future.
  if (dstMaskValues != None):
^@ OBS SHAPE IS  (12, 72, 144)
 sos   GFDL-CM3   (12, 72, 144)   (12, 72, 144)    JPL-Aquarius-v2
 Error while processing variable sos:
	not all arguments converted during string formatting
ok

======================================================================
FAIL: test_from_parameter_file (test_from_param.TestFromParam)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/data/jservon/PMP_installs/2015-11-16/PCMDI_METRICS/tmp/metrics/test/test_from_param.py", line 68, in test_from_parameter_file
    self.assertTrue(ok)
AssertionError: False is not true

----------------------------------------------------------------------
Ran 8 tests in 89.206s

FAILED (failures=1)
```
And meanwhile the "default" install (with cdat_version="2.1.0" and metrics_checkout="v1.0.0") works fine.
I'm a little bit confused... any idea?

**Comment by durack1** [comment]
@jservonnat yep this is expected.. and was what I meant by the "tweaks which should be merged into the master branch" comment [above](https://github.com/PCMDI/pcmdi_metrics/issues/288#issuecomment-155657277)

If you can wait a week (@doutriaux1 is locked up finalising the UV-CDAT 2.4.0 release) these issues in `master` should be solved soon for your reference when #279 and #277 are closed this should be solved..
